### PR TITLE
refactor(github-actions): Format and rename workflow jobs

### PR DESCRIPTION
**Pull request by queengooborg**
This performs some formatting of the GitHub Actions workflows to help ensure more standard-looking formatting.  Additionally, this renames a few workflow jobs so they don't all have the same name.

Note: I found we have three labeler workflows (minus the rebase-needed workflow), maybe we should consolidate them?

### CODE REVIEW

**Comment by teoli2003** [comment]
cc/ing @bsmth who always has good ideas (strong ideas?) about naming scripts 😄

**Comment by bsmth** [comment]
Thanks @teoli2003 🙌🏻 I'll take a look -> assigned myself so I don't forget.

File: .github/workflows/pr-check_javascript.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
       - .github/workflows/pr-check_javascript.yml
 
 jobs:
-  docs:
+  lint-js:
```
Comment (bsmth) [inline .github/workflows/pr-check_javascript.yml:14]: This has slightly annoyed me for so long, glad this is taken care of, thank you 😄

**Review by bsmth — APPROVED** [review]
Looks good, thanks. 

- [x] Checked if we use `needs:` anywhere [for prerequisite jobs](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs) so we can rename safely.

>Note: I found we have three labeler workflows (minus the rebase-needed workflow), maybe we should consolidate them?

I think we have too many workflows, so it would be good to reduce this where possible. Can we do this in a follow-up?

**Pull request merged**